### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Sites.Closed

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mo271** [comment]
TODO:

- [ ] `Functor.closedSieves`: seems aesop is not able to fill the missing `_`.
- [ ] `classifier_isSheaf`: `simp_rw [← c]` doesn't to the intended subsitution, probably because a type is involved.
- [ ] naming in comments

**Comment by int-y1** [comment]
The error with `Functor.closedSieves` no longer exists

File: Mathlib/CategoryTheory/Sites/Closed.lean [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+-/
+def IsClosed {X : C} (S : Sieve X) : Prop :=
+  ∀ ⦃Y : C⦄ (f : Y ⟶ X), J₁.Covers S f → S f
+#align category_theory.grothendieck_topology.is_closed CategoryTheory.GrothendieckTopology.IsClosed
+
+/-- If `S` is `J₁`-closed, then `S` covers exactly the arrows it contains. -/
+theorem covers_iff_mem_of_closed {X : C} {S : Sieve X} (h : J₁.IsClosed S) {Y : C} (f : Y ⟶ X) :
+    J₁.Covers S f ↔ S f :=
+  ⟨h _, J₁.arrow_max _ _⟩
+#align category_theory.grothendieck_topology.covers_iff_mem_of_closed CategoryTheory.GrothendieckTopology.covers_iff_mem_of_closed
```
Comment (semorrison) [inline Mathlib/CategoryTheory/Sites/Closed.lean:89]:
```suggestion
theorem covers_iff_mem_of_isClosed {X : C} {S : Sieve X} (h : J₁.IsClosed S) {Y : C} (f : Y ⟶ X) :
    J₁.Covers S f ↔ S f :=
  ⟨h _, J₁.arrow_max _ _⟩
#align category_theory.grothendieck_topology.covers_iff_mem_of_closed CategoryTheory.GrothendieckTopology.covers_iff_mem_of_isClosed
```
Perhaps take the opportunity to fix the name.

File: Mathlib/CategoryTheory/Sites/Closed.lean [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+    change J₁.close S f
+    rwa [h]
+  · intro hS
+    rw [eq_top_iff]
+    intro Y f _
+    apply J₁.pullback_stable _ hS
+#align category_theory.grothendieck_topology.close_eq_top_iff_mem CategoryTheory.GrothendieckTopology.close_eq_top_iff_mem
+
+/-- A Grothendieck topology induces a natural family of closure operators on sieves. -/
+-- @[simps (config := { rhsMd := semireducible })] Porting note: no semireducible
```
Comment (semorrison) [inline Mathlib/CategoryTheory/Sites/Closed.lean:169]:
```suggestion
```
We're making this replacement everywhere, and not leaving a porting note. I think it's okay to do so here, too.

**Comment by semorrison** [comment]
bors d+

**Pull request closed without merging**